daemon/logger/local/local.go

```diff
@@ -149,7 +147,15 @@ func (d *driver) Name() string {
 }
 
 func (d *driver) Log(msg *logger.Message) error {
-	return d.logfile.WriteLogEntry(msg)
+	defer logger.PutMessage(msg)
```

Comment:
Optimization opportunity: `msg` can be returned to the pool as soon as `marshal` returns, rather than deferring until after `WriteLogEntry`. Be mindful of that timestamp, though!